yt/yt/server/controller_agent/controllers/operation_controller_detail.cpp

```diff
@@ -10035,10 +10035,27 @@ void TOperationControllerBase::InitUserJobSpecTemplate(
     jobSpec->set_rpc_proxy_worker_thread_pool_size(jobSpecConfig->RpcProxyWorkerThreadPoolSize);
 
     // Pass external docker image into job spec as is.
-    if (jobSpecConfig->DockerImage &&
-        !TDockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry).IsInternal())
+    if (jobSpecConfig->DockerImage)
     {
-        jobSpec->set_docker_image(*jobSpecConfig->DockerImage);
+        if (TDockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry).IsInternal()) {
+            if (!Config->Environment.contains("YT_SECURE_VAULT_docker_auth") && !jobSpecConfig->Environment.contains("YT_SECURE_VAULT_docker_auth")) {
```

Comment:
Let's make `YT_SECURE_VAULT_docker_auth` const somewhere.

Reply:
I have avoided the duplication using a lambda. Is it good enough now?